File: app/src/main/java/com/esoxjem/movieguide/listing/sorting/SortingDialogInteractorImpl.java
package com.esoxjem.movieguide.listing.sorting;

/**
 * @author arun
 */
class SortingDialogInteractorImpl implements SortingDialogInteractor {
    private SortingOptionStore sortingOptionStore;

    SortingDialogInteractorImpl(SortingOptionStore store) {
        sortingOptionStore = store;
    }

    @Override
    public int getSelectedSortingOption() {
        return sortingOptionStore.getSelectedOption();
    }

    @Override
    public void setSortingOption(SortType sortType) {
        sortingOptionStore.setSelectedOption(sortType);
    }
}

File: app/src/main/java/com/esoxjem/movieguide/movies/sorting/SortingDialogInteractorImpl.kt
package com.esoxjem.movieguide.movies.sorting

/**
 * @author arunsasidharan
 */
internal class SortingDialogInteractorImpl(private val sortingOptionStore: SortingOptionStore) : SortingDialogInteractor {

    override fun getSelectedSortingOption() = sortingOptionStore.getSelectedOption()

    override fun setSortingOption(sortType: SortType) = sortingOptionStore
            .setSelectedOption(sortType)
}
